gapic/templates/%namespace/%name_%version/%sub/services/%service/async_client.py.j2

```diff
@@ -599,6 +599,12 @@ class {{ service.async_client_name }}:
         return response
     {% endif %}
 
+    async def __aenter__(self):
+        return self
+
+    async def __aexit__(self, exc_type, exc, tb):
+        await self.transport.close()
```

Comment:
Are we certain the `transport` will exist at the time of exiting? Same for other `aexits`.

Reply:
Yes, because it gets created in `__init__`. @software-dov correct me if Im wrong.

Reply:
Yes, it should. The client will construct a transport in `__init__` if one isn't passed in. The user is of course free to set the field manually or create a client that breaks assumptions, but this should be safe.